### PR TITLE
feat: custom messages part 2

### PR DESCRIPTION
**Pull request by UberWaffe**
Allow for map maker defined custom messages that can be used in the scenario and events.

### CODE REVIEW

File: src/window/message_dialog.c [hunk 1]
```diff
@@ -118,18 +131,103 @@ static void set_city_message(int year, int month,
...
+    } else {
+        return &data.custom_lang_message;
+    }
+}
+
+static void init(int text_id, int is_custom_message, void (*background_callback)(void))
 {
     scroll_drag_end();
+    sound_speech_stop();
+    sound_music_stop();
```
Comment (crudelios) [inline src/window/message_dialog.c:216]: Won't these two lines stop playing music even if no custom music is set?

File: src/window/message_dialog.c [hunk 2]
```diff
@@ -118,18 +131,103 @@ static void set_city_message(int year, int month,
...
+{
+    if (!data.show_video) {
+        if (data.should_play_audio) {
+            sound_speech_play_file(custom_messages_get_audio(data.custom_msg));
+        }
+        if (data.should_play_background_music) {
+            sound_device_stop_music();
+            int volume = setting_sound(SOUND_MUSIC)->volume;
+            sound_device_play_music(custom_messages_get_background_music(data.custom_msg), volume);
+        }
```
Comment (crudelios) [inline src/window/message_dialog.c:189]: In the future, I want a setting like `music_as_speech_background` that plays at a lower volume and fades out the music when the speech ends.

**Review by crudelios — APPROVED** [review]
LGTM, just some small questions.

File: src/window/message_dialog.c (hunk 2 above)
Comment (UberWaffe) [reply src/window/message_dialog.c:189]: I'm not sure how to do that.
I checked the existing windows that had to do with introductions (as far as I could tell) and didn't see anything like that.
Did I miss it somewhere?

File: src/window/message_dialog.c (hunk 1 above)
Comment (UberWaffe) [reply src/window/message_dialog.c]: You are correct.
Fixed.

File: src/window/message_dialog.c (hunk 2 above)
Comment (UberWaffe) [reply src/window/message_dialog.c:189]: I completely misunderstood then when you first mentioned this in Discord.
I thought that was in code and already a feature, but I understand now you meant it was baked into the audio files.
When I didn't see it in code I assumed it is somehow baked into the `sound_device_play_music` code and I was just not seeing it.

I'll see if I can do a follow-up PR in the near future to include this feature.

**Comment by UberWaffe** [comment]
Rebased.

**Pull request merged**